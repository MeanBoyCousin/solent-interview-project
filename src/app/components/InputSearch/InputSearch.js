import React from 'react'
import debounce from 'lodash.debounce'

import { InputSearchWrapper } from './InputSearch.styled'

const InputSearch = ({ setQuery }) => {
    const debounceSetQuery = debounce(e => {
        setQuery(e.target.value.toLowerCase())
    }, 300)

    return (
        <InputSearchWrapper>
            <img
                src={`${process.env.ICONS}search.svg`}
                alt="A search icon indicating that this input box is for searching the page content"
            />
            <input
                type="text"
                placeholder="Start typing..."
                aria-label="Search"
                onKeyUp={e => {
                    e.persist()
                    debounceSetQuery(e)
                }}
            />
        </InputSearchWrapper>
    )
}

export { InputSearch }
