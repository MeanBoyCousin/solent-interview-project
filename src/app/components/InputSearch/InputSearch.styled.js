import styled from 'styled-components'

const InputSearchWrapper = styled.div`
    width: 85%;
    position: relative;
    margin-bottom: 10%;

    img {
        position: absolute;
        top: 0;
        right: 0;
        height: 100%;
        transform: scale(0.7);
    }

    input {
        box-sizing: border-box;
        width: 100%;
        padding: 5%;
        border: none;
        border-radius: 10px;

        &::placeholder {
            opacity: ${props => props.theme.placeholder};
        }
    }

    @media (min-width: 768px) {
        img {
            transform: scale(0.6);
        }

        input {
            font-size: 1.25rem;
            padding: 2.5%;
        }
    }

    @media (min-width: 992px) {
        width: 50%;
        position: relative;
        margin-bottom: 5%;

        img {
            transform: scale(0.5);
        }

        input {
            font-size: 1.5rem;
            padding: 2.5%;
        }
    }
`

export { InputSearchWrapper }
