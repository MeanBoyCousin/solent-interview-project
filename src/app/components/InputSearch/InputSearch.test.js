import React from 'react'
import 'dotenv/config'
import '@testing-library/jest-dom'
import { render, cleanup } from '@testing-library/react'

import { InputSearch } from './InputSearch'

afterEach(cleanup)

describe('InputSearch component', () => {
    test('should render input field of type text with search icon', () => {
        const { asFragment } = render(<InputSearch />)
        expect(asFragment()).toMatchSnapshot()
    })
})
