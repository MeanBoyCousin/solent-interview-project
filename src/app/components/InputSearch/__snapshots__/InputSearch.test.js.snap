// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputSearch component should render input field of type text with search icon 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa jChFM"
  >
    <img
      alt="A search icon indicating that this input box is for searching the page content"
      src="/static/icons/search.svg"
    />
    <input
      aria-label="Search for a member of staff"
      placeholder="Start typing..."
      type="text"
    />
  </div>
</DocumentFragment>
`;
